addons/binding/org.openhab.binding.http/src/main/java/org/openhab/binding/http/handler/HttpThingHandler.java

```diff
@@ -0,0 +1,405 @@
+        } else if (!this.commandRequest.isPresent()) {
+            logger.warn("[{}] got command on channel '{}', but no command URL set", getThing().getUID(), channelUID.getId());
+        } else {
+            final CommandRequest commandRequest = this.commandRequest.get();
+            final CommandRequest.Method method = commandRequest.getMethod();
+            final String commandStr = command.toFullString();
+            try {
+                final String transformedCommand = doTransform(commandRequest.getRequestTransform(), commandStr);
+                final URL transformedUrl = formatUrl(commandRequest.getUrl(), transformedCommand);
+                makeHttpRequest(method, transformedUrl, commandRequest.getContentType(), Optional.empty(), Optional.of(commandStr)).whenComplete((response, t) -> {
+                    if (t != null) {
+                        updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR, "Connetion to server failed when sending command: " + t.getMessage());
+                    } else if (response.getResponse().getStatus() / 100 != 2) {
+                        updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR, "Server returned HTTP status " + response.getResponse().getStatus() + " when sending command");
+                    } else {
+                        updateStatus(ThingStatus.ONLINE);
+                        updateState(stateFromResponse(response, commandRequest.getResponseTransform()));
+                    }
+                });
+            } catch (final IllegalArgumentException e) {
+                updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR, e.getMessage());
+            }
+        }
+    }
+
+    @Override
+    protected void updateStatus(final ThingStatus status, final ThingStatusDetail statusDetail, @Nullable final String description) {
+        final ThingStatusInfo curStatusInfo = getThing().getStatusInfo();
+        if (!Objects.equals(status, curStatusInfo.getStatus()) || !Objects.equals(statusDetail, curStatusInfo.getStatusDetail()) || !Objects.equals(description, curStatusInfo.getDescription())) {
+            super.updateStatus(status, statusDetail, description);
+        }
+    }
+
+    @Override
+    public void handleConfigurationUpdate(final Map<String, Object> configurationParameters) {
+        super.handleConfigurationUpdate(configurationParameters);
+        try {
+            updateConfig(configurationParameters);
+            updateStatus(ThingStatus.ONLINE);
+        } catch (final IllegalArgumentException e) {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR, e.getMessage());
+        }
+    }
+
+    @Override
+    public void channelLinked(ChannelUID channelUID) {
+        if (channelUID.equals(getThing().getChannel(CHANNEL_STATE).getUID())) {
+            this.linkedItems = this.itemChannelLinkRegistry.getLinkedItems(channelUID);
```

Comment:
Nice idea, I see where you want to go with this. But please don't keep references in a handler to items. Use the method `isChannelLinked` or so (I can't recall exactly) in your `updateState` method.